substrate/client/network/benches/request_response_protocol.rs

```diff
@@ -119,71 +111,115 @@ where
 		notification_metrics: NotificationMetrics::new(None),
 	})
 	.unwrap();
+	let notification_service = Arc::new(Mutex::new(notification_service));
+	let network_service = worker.network_service();
 
-	(worker, rx, notification_service)
+	(worker, network_service, rx, notification_service)
 }
 
-async fn run_serially<B, H, N>(size: usize, limit: usize)
+struct BenchSetup {
+	#[allow(dead_code)]
+	notification_service1: Arc<Mutex<Box<dyn NotificationService>>>,
+	#[allow(dead_code)]
+	notification_service2: Arc<Mutex<Box<dyn NotificationService>>>,
+	network_service1: Arc<dyn NetworkService>,
+	peer_id2: PeerId,
+	handle1: JoinHandle<()>,
+	handle2: JoinHandle<()>,
+	#[allow(dead_code)]
+	rx1: async_channel::Receiver<IncomingRequest>,
+	rx2: async_channel::Receiver<IncomingRequest>,
+}
+
+impl Drop for BenchSetup {
+	fn drop(&mut self) {
+		self.handle1.abort();
+		self.handle2.abort();
+	}
+}
+
+fn setup_workers<B, H, N>(rt: &tokio::runtime::Runtime) -> Arc<BenchSetup>
 where
 	B: BlockT<Hash = H256> + 'static,
 	H: ExHashT,
 	N: NetworkBackend<B, H>,
 {
-	let listen_address1 = get_listen_address();
-	let listen_address2 = get_listen_address();
-	let (worker1, _rx1, _notification_service1) = create_network_worker::<B, H, N>(listen_address1);
-	let service1 = worker1.network_service().clone();
-	let (worker2, rx2, _notification_service2) =
-		create_network_worker::<B, H, N>(listen_address2.clone());
+	let _guard = rt.enter();
```

Comment:
What is this needed for?

Reply:
Without it we can't initialize workers that needs to spawn a tokio task inside.